src/utils/tableUtils.ts

```diff
@@ -0,0 +1,167 @@
+import JSZip from "jszip";
+import { ColumnMetadata, TableData, dataTypes } from "../types";
+import { defaults, element, elementAttributes, queryTableNotFoundErr, queryTableXmlPath, sheetsNotFoundErr, sheetsXmlPath, tableNotFoundErr, tableXmlPath, textResultType, workbookXmlPath, xmlTextResultType } from "../constants";
+import documentUtils from "./documentUtils";
+import { v4 } from "uuid";
+
+const updateTableInitialDataIfNeeded = async (zip: JSZip, tableData?: TableData) : Promise<void> => {
+        if (!tableData)
+        {
+            return;
+        }
+
+        const sheetsXmlString: string | undefined = await zip.file(sheetsXmlPath)?.async(textResultType);
+        if (sheetsXmlString === undefined) {
+            throw new Error(sheetsNotFoundErr);
+        }
+
+        const newSheet: string = await updateSheetsInitialData(sheetsXmlString, tableData);
+        zip.file(sheetsXmlPath, newSheet);
+
+        const queryTableXmlString: string | undefined = await zip.file(queryTableXmlPath)?.async(textResultType);
+        if (queryTableXmlString === undefined) {
+            throw new Error(queryTableNotFoundErr);
+        }
+
+        const newQueryTable: string = await updateQueryTablesInitialData(queryTableXmlString, tableData);
+        zip.file(queryTableXmlPath, newQueryTable);
+
+        const tableXmlString: string | undefined = await zip.file(tableXmlPath)?.async(textResultType);
+        if (tableXmlString === undefined) {
+            throw new Error(tableNotFoundErr);
+        }
+
+        const newTable: string = await updateTablesInitialData(tableXmlString, tableData);
+        zip.file(tableXmlPath, newTable);
+
+        const workbookXmlString = await zip.file(workbookXmlPath)?.async(textResultType);
+        if (workbookXmlString === undefined) {
+            throw new Error(sheetsNotFoundErr);
+        }
+        
+        const newWorkbook:string = await updateWorkbookInitialData(workbookXmlString, tableData);
+        zip.file(workbookXmlPath, newWorkbook);
+    };
+
+const updateTablesInitialData = async (tableXmlString: string, tableData: TableData) : Promise<string> => {
+        const parser: DOMParser = new DOMParser();
+        const serializer: XMLSerializer = new XMLSerializer();
+        const tableDoc: Document = parser.parseFromString(tableXmlString, xmlTextResultType);
+        const tableColumns: Element = tableDoc.getElementsByTagName(element.tableColumns)[0];
```

Comment:
We are not validate anything when we call to all the getElemantByTagName in this file, are we sure all of them must exist?